### ISSUE TITLE
Warning: using box for unknown shape cylinder

### ISSUE THREAD

**Issue by volkf** [ellson/graphviz#84]
`Warning: using box for unknown shape cylinder`

Always getting this error with version 2.38...will there be an updated package?

**Comment by russtoku** [comment]
On Mac OS X 10.10.5 (Yosemite), I got the same error after installing with "brew install graphviz". The HEAD (81fc9e8) works when installing "brew install --HEAD graphviz" but needs to have autogen.sh fixed so that "AR_FLAGS=cr" instead of "AR_FLAGS=crD". dot says "graphviz version 2.39.20161205.1853".

**Comment by ellson** [comment]
I've removed the over-ride.  Hopefully the system defaults will be correct.

How did this get to be reported against issue #84 ?

John

**Comment by russtoku** [comment]
After I ran into the problem with the default Homebrew formula for graphviz (version 2.38.0), I searched the issues for "Warning: using box for unknown shape cylinder" and this issue popped up. Hope that's what you were asking about.

**Comment by ellson** [comment]
No problem,  but it would be better next time if you would create a new issue the topic is different. 

       https://github.com/ellson/graphviz/issues


Thanks very much for the report.  I've committed a change which I hope fixes it.  If not, please feel free to reply to this thread, this time :-)


John

**Comment by russtoku** [comment]
Apologies for piggy-backing on this issue. I assumed that the issue's title was close enough. Now I see from your comment and a re-read of the original comment how I was mistaken. While I experienced the same error message, my first comment didn't address the original comment's question of when would there be a newer release. With your feedback, I can see that this issue's title doesn't match with "fix" that I suggested. My "fix" addresses the problem of HEAD not compiling on Macs. A new issue with a title something like "HEAD doesn't compile on Mac OS X" would be more useful. Especially, given the comment in autogen.sh about the "D" option being added.

Thanks for the feedback and useful tool.

**Comment by skelter** [comment]
Running into this with /usr/local/Cellar/graphviz/2.40.1..

Brewing from head fails:
$  brew install --HEAD graphviz
...
Error: inreplace failed
lib/sfio/features/sfio:
  expected replacement of "lib qfrexp\nlib qldexp\n" with ""

**Comment by ellson** [comment]
Are you geting a build error, or a runtime error?  I'm confused.

I've no idea what "lnreplace" is, and there is no "lib qfrexp" in "lib/sfio/features/sfio"


The command:
     echo "graph{a [shape=cylinder]}" | dot -Tx11

works for me with:  2.41.20170215.1739

(Although,  something has changed recently with Fedora dnf rpm install ordering, so I needed to run "sudo dot -c" after install.    Hopefully I've got this fixed in the next build.)

**Comment by skelter** [comment]
Elison, I am getting that error when I run $ brew install --HEAD graphviz.  full ouput is

``````
$ brew install --HEAD graphviz
==> Using the sandbox
==> Cloning https://github.com/ellson/graphviz.git
Updating /Users/steve.suehs/Library/Caches/Homebrew/graphviz--git
==> Checking out branch master
Synchronizing submodule url for 'dependencies/boxfort'
Synchronizing submodule url for 'dependencies/debugbreak'
Synchronizing submodule url for 'dependencies/dyncall'
Synchronizing submodule url for 'dependencies/klib'
Synchronizing submodule url for 'dependencies/libcsptr'
Synchronizing submodule url for 'dependencies/nanomsg'
Synchronizing submodule url for 'dependencies/nanopb'
Synchronizing submodule url for 'dependencies/wingetopt'
Error: inreplace failed
lib/sfio/features/sfio:
  expected replacement of "lib qfrexp\nlib qldexp\n" with ""
$ uname -a
Darwin C02R4239G8WM.local 15.6.0 Darwin Kernel Version 15.6.0: Mon Jan  9 23:07:29 PST 2017; root:xnu-3248.60.11.2.1~1/RELEASE_X86_64 x86_64
``````

**Comment by ellson** [comment]
The "lib qfrexp" reference was removed last month (commit# 187898e08e74a1f0fa3bc807aeca9004e1342c77)

It sounds like you have some "brew": script that is not a part of graphviz, that needs to be updated.

And please stop piggy-backing these bug reports.

**Comment by ellson** [comment]
Not a graphviz bug.

**Comment by cleberjamaral** [comment]
On linux debian stable version 2.38.0-17, the cylinder is not working. It may be fixed using buster (testing) release 2.40.1-5+b1.

However, another problem I've noticed on linux is about URLs, even with last available version, links to websites do not work.

**Comment by ellson** [comment]
The repository for graphviz has been moved to
https://gitlab.com/graphviz/graphviz

The code on github is stale and no longer maintained.

The home page is at https://graphviz.org/

Coulf you provide more details on which URLs are not working?

John

**Comment by cleberjamaral** [comment]
Hi John,

It was my mistake.

To fix the problem I moved from <img src=\"mysvg.svg\"> to <object data=\"mysvg.svg\" type=\"image/svg+xml\"></object> what makes url links work well. More details: https://stackoverflow.com/questions/4476526/do-i-use-img-object-or-embed-for-svg-files and https://stackoverflow.com/questions/7008355/how-to-make-links-in-an-embedded-svg-file-open-in-the-main-window-not-in-a-sepa

guru.nidi:graphviz-java:0.7.0 and target tag, are also working pretty well.

I have compiled the last snapshot available on graphviz web site (graphviz version 2.41.20171026.1811 (20171026.1811)) but I don't think it is necessary because svg file created by "2.40.1-5+b1" has xlink hrefs.

Thanks for your attention!
